### ISSUE TITLE
[FLANG][OpenMP] flang fails to produce appropriate error message for array section in depend clause

### ISSUE THREAD

**Issue by laoshd** [llvm/llvm-project#225225]
According to Restrictions of OpenMP 5.0 2.17.11 depend Clause:
```
• Array sections cannot be specified in depend clauses with the depobj dependence type.
• List items used in depend clauses with the depobj dependence type must be depend objects in the initialized state.
```
flang should show a better error message than "not yet implemented: DEPOBJ dependence-type" for the this simple test:

`% cat tst.f90`

```fortran
integer :: x(10)
!$omp task depend(depobj: x(:))
!$omp end task
end
```

```console
% flang -fopenmp -fopenmp-version=50 tst.f90
warning: OpenMP support for version 50 in flang is still incomplete [-Wexperimental-option]
error: loc("/home/users/laos/llvm-flang/test/CPE-18454/tst.f90":2:7): /home/users/laos/llvm-flang/src/llvm-project_pe31/flang/lib/Lower/OpenMP/ClauseProcessor.cpp:200: not yet implemented: DEPOBJ dependence-type
LLVM ERROR: aborting
```

**Comment by laoshd** [comment]
I'm interested in working on this. Would you please help to assign it to me (@laoshd)? Thanks.
